db/migrate/20250625120000_fix_broken_events.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2025, Pfadibewegung Schweiz. This file is part of
+#  hitobito_pbs and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_pbs.
+
+class FixBrokenEvents < ActiveRecord::Migration[6.1]
+  def up
+    still_invalid = {}
+    say_with_time "cleaning up invalid events" do
+      Event.includes(:questions, :groups).find_each do |event|
+        next if event.valid? && event.questions.all?(&:valid?)
+
+        event.required_contact_attrs&.reject! { |a| a == 'address' }
+        event.required_contact_attrs&.reject! { |a| a == 'correspondence_language' }
+        event.hidden_contact_attrs&.reject! { |a| a == 'address' }
+        event.hidden_contact_attrs&.reject! { |a| a == 'correspondence_language' }
+        event.contact_attrs_passed_on_to_supercamp&.reject! { |a| a == 'address' }
+        event.contact_attrs_passed_on_to_supercamp&.reject! { |a| a == 'correspondence_language' }
+
+        clean_up_duplicate_questions!(event)
+
+        event.save!(validate: false) # save without validations, to ignore any remaining user errors
+      rescue ActiveRecord::RecordInvalid
+        still_invalid[event.id] = event.errors
+      end
```

Comment:
This will only be reached by invalid answers as we ignore the validation on the events. So, we see the event that has invalid answers but not the errors related to that, right?

Am I missing something?

Reply:
Ooops, nice catch. Das `validate: false` kam erst spät dazu, weil ich gemerkt habe dass die Fixes auch dann gespeichert werden sollen wenn es noch andere, userverschuldete Fehler hat. Statt dem rescue wäre jetzt wohl eher ein `unless event.valid?` angebracht.